tests/test_ext_autodoc_configs.py

```diff
@@ -1584,6 +1584,14 @@ def test_autodoc_typehints_format_fully_qualified_for_newtype_alias(app):
 
 @pytest.mark.sphinx('html', testroot='ext-autodoc')
 def test_autodoc_default_options(app):
+    if (
+            (3, 11, 7) <= sys.version_info < (3, 12)
+            or sys.version_info >= (3, 12, 1)
+    ):
+        list_of_weak_references = "      list of weak references to the object"
+    else:
+        list_of_weak_references = "      list of weak references to the object (if defined)"
```

Comment:
```suggestion
    list_of_weak_references = "      list of weak references to the object"
```

Reply:
This won't work, because in `assert list_of_weak_references in actual`, `actual` is a list of strings.

Python 3.12.1:

```python
[
    "",
    ".. py:class:: CustomIter()",
    "   :module: target",
    "",
    "",
    "   .. py:method:: CustomIter.__init__()",
    "      :module: target",
    "",
    "      Create a new `CustomIter`.",
    "",
    "",
    "   .. py:method:: CustomIter.__iter__()",
    "      :module: target",
    "",
    "      Iterate squares of each value.",
    "",
    "",
    "   .. py:attribute:: CustomIter.__weakref__",
    "      :module: target",
    "",
    "      list of weak references to the object",
    "",
    "",
    "   .. py:method:: CustomIter.snafucate()",
    "      :module: target",
    "",
    "      Makes this snafucated.",
    "",
]
```

Python 3.10.13:

```python
[
    "",
    ".. py:class:: CustomIter()",
    "   :module: target",
    "",
    "",
    "   .. py:method:: CustomIter.__init__()",
    "      :module: target",
    "",
    "      Create a new `CustomIter`.",
    "",
    "",
    "   .. py:method:: CustomIter.__iter__()",
    "      :module: target",
    "",
    "      Iterate squares of each value.",
    "",
    "",
    "   .. py:attribute:: CustomIter.__weakref__",
    "      :module: target",
    "",
    "      list of weak references to the object (if defined)",
    "",
    "",
    "   .. py:method:: CustomIter.snafucate()",
    "      :module: target",
    "",
    "      Makes this snafucated.",
    "",
]
```